### PR TITLE
add createUser and login route and controller

### PR DESCRIPTION
**Pull request by lolerwaffles**

### CODE REVIEW

File: backend/routes.js [hunk 1]
```diff
@@ -1,17 +1,27 @@
...
-    } catch(error) {
-        res.status(400).json({error: 'something went wrong'});
-    }
+router.post('/newUser', async (req, res) => {
+  try {
+    const { username, email, password } = req.body;
+    const encryptedPassword = await bcrypt.hash(password, saltRounds);
+    const newUser = new User({ username, email, password: encryptedPassword });
+    const savedUser = await newUser.save();
+    res.status(201).json(savedUser);
```
Comment (bethanyann) [inline backend/routes.js:13]: Can you remove this code that hashes the password and saves the user in the DB from the routes file and instead call the createUser function like you're doing for the login route down below, or is there a reason you included db functions in this file as well as duplicating this code in the userController?

File: backend/controllers/userController.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
+const mongoose = require('mongoose');
+const User = require('../models/userModel.js');
+
+const bcrypt = require('bcrypt');
+const saltRounds = 12; // you can adjust this value as needed
+
+const createUser = async (req, res) => {
+  try {
+    const { username, email, password } = req.body;
+    const encryptedPassword = await bcrypt.hash(password, saltRounds);
```
Comment (bethanyann) [inline backend/controllers/userController.js:10]: Nothing wrong here, mostly just a nitpick to make the destructuring of the req.body params either both inside the try block or outside of it between the login and createUser functions for code consistency between the two functions.

File: backend/routes.js [hunk 3]
```diff
@@ -1,17 +1,27 @@
...
 
-// sample route
-// a  route like this would fire
-router.get('/', (req, res) => {
-    try {
-        res.json({msg: 'test GET request working'});
-    } catch(error) {
-        res.status(400).json({error: 'something went wrong'});
-    }
+router.post('/newUser', async (req, res) => {
```
Comment (bethanyann) [inline backend/routes.js]: This isn't the correct endpoint mentioned in the task. The frontend has been set up to have the endpoint "/api/signup" please make this endpoint reflect that

File: backend/routes.js [hunk 4]
```diff
@@ -1,17 +1,27 @@
...
+    const encryptedPassword = await bcrypt.hash(password, saltRounds);
+    const newUser = new User({ username, email, password: encryptedPassword });
+    const savedUser = await newUser.save();
+    res.status(201).json(savedUser);
+  } catch (error) {
+    res.status(400).json({ message: error.message });
+  }
 });
 
+router.post('/login', login);
```
Comment (bethanyann) [inline backend/routes.js:19]: this endpoint needs to be "/api/login" to match the route pattern

Comment (lolerwaffles) [reply backend/routes.js:19]: /api/ is implied from the server.js, tested.

File: backend/routes.js (hunk 3 above)
Comment (lolerwaffles) [reply backend/routes.js]: fixed, thank you

File: backend/routes.js (hunk 1 above)
Comment (lolerwaffles) [reply backend/routes.js:13]: removed, ty. Guess I was double hashing

**Review by bethanyann — APPROVED** [review]
looks great!

**Pull request merged**